pkg/protocol/engine/ledger/ledger/ledger.go

```diff
@@ -225,7 +228,24 @@ func (l *Ledger) Output(outputID iotago.OutputID) (*utxoledger.Output, error) {
 	case *ExecutionOutput:
 		txWithMetadata, exists := l.memPool.TransactionMetadata(outputID.TransactionID())
 		if !exists {
-			return nil, err
+			var output *utxoledger.Output
+			stateRequest := l.resolveState(outputID.UTXOInput())
```

Comment:
why are you resolving here? this should come from the mempool. If it is for the API, look at the other function I added `OutputOrSpent`

Reply:
If the mempool happens to still hold this "intermediate" representation of an executed output you might run into a race condition where the transaction metadata is not available anymore in the mempool to create an utxoledger.Output. If you run into this race then you try to resolve the output against the storage, and at some point it must be there, as the race resolves: hence the promise.